pkg/common/k8s_utils_test.go

```diff
@@ -359,3 +371,527 @@ func TestIsObjectTaggedToDelete(t *testing.T) {
 		})
 	}
 }
+
+func TestStatusConditionsMarshalJSON(t *testing.T) {
+	now := time.Now()
+	nowFmt := now.UTC().Format("2006-01-02T15:04:05Z")
+
+	testCases := []struct {
+		name     string
+		input    []metav1.Condition
+		expected []byte
+		err      error
+	}{
+		{
+			name:     "when input is empty then return an empty JSON (empty byte array)",
+			input:    []metav1.Condition{},
+			expected: []byte("[]"),
```

Comment:
`"[]"` is not an empty byte array; it's a byte array of length 2 (1 byte per character, e.g. if ASCII-encoded or ASCII compatible-encoded).